src/components/tagsInput/index.js

```diff
@@ -198,6 +207,14 @@ export default class TagsInput extends BaseComponent {
     return isLastTagMarked;
   }
 
+  updateInvalidTags(tags) {
```

Comment:
instead of going through all of the tags to search if there's an invalid tag. 
You can determine if you are in an invalid state by checking if the user passed the error message prop. (what you call `onTagNotValidate`) 
You don't even need to save the `isTagNotValidate` in your state, you can base everything on the prop the user pass.

Reply:
but what if the user is passing the error message , but not passing an invalid tag in tags,
an error message shouldn't appear ?

Reply:
It should.. 
the `validationErrorMessage` controls the error message
the `tag.invalid` controls the tag error UI

if the user didn't pass both it's his problem 
Also, it's possible we want to show an error message without marking any tag as invalid .
for example, if I allow only 5 tags and user add 6 tag. 
In that case I will **only** pass the error message without marking any tag as invalid.

Reply:
ok , done